keps/sig-api-machinery/4192-svm-in-tree/README.md

```diff
@@ -0,0 +1,484 @@
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/controller.go:127:        enqueueResource                         100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/controller.go:136:        Run                                     0.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:36:  processDiscovery                        79.2%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:75:  toGroupResource                         100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:84:  cleanMigrations                         75.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:107: launchMigration                         100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:121: relaunchMigration                       66.7%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:129: newStorageState                         100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:143: updateStorageState                      71.4%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:181: staleStorageState                       100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:185: processDiscoveryResource                66.7%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:219: isMigrated                              100.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/discovery_handler.go:226: hasPendingOrRunningMigration            80.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/migration_handler.go:33:  storageStateName                        66.7%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/migration_handler.go:42:  markStorageStateSucceeded               0.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/migration_handler.go:68:  processMigration                        0.0%
+sigs.k8s.io/kube-storage-version-migrator/pkg/trigger/migration_handler.go:83:  processQueue                            0.0%
+total:                                                                          (statements)                            45.9%
+```
+
+##### Integration tests
+
+- [ ] Write a test which will identify the migratable resource, create Migration resource for the same and test whether Storage Version Migration is actually carried out.
+
+##### e2e tests
+
+- [ ] Test to validate functionality of the Migration controller.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Resolve potential kube-apiserver OOMs and cascading failure issues that may occur due to the use of paginated lists.
+- Feature is enabled by default
+- All of the above documented tests are complete
+
+### Upgrade / Downgrade Strategy
```

Comment:
Can you please fill in upgrade/downgrade statrgy and version skew stragegy?

Reply:
Thanks @wojtek-t. I opened https://github.com/kubernetes/enhancements/pull/4276 to address this.